test/dotnet-new3.UnitTests/PostActionTests.cs

```diff
@@ -214,6 +214,89 @@ public void RunScript_Basic()
+                .WithCustomHive(home)
+                .WithWorkingDirectory(workingDirectory)
+                .Execute();
+
+            commandResult
+                .Should()
+                .ExitWith(0)
+                .And.HaveStdOutContaining($"The template \"{templateName}\" was created successfully.")
+                .And.HaveStdOutContaining("This line goes to stdout")
+                .And.HaveStdErrContaining("This line goes to stderr")
+                .And.NotHaveStdOutContaining("Run 'chmod +x *.sh'")
+                .And.NotHaveStdOutContaining("Manual instructions: Run 'setup.cmd'")
+                .And.NotHaveStdOutContaining("Manual instructions: Run 'setup.sh'");
+        }
+
+        [Fact]
+        public void RunScript_Redirect()
+        {
+            string templateLocation = "PostActions/RunScript/Redirect";
+            string templateName = "TestAssets.PostActions.RunScript.Redirect";
+            string home = TestUtils.CreateTemporaryFolder("Home");
+            string workingDirectory = TestUtils.CreateTemporaryFolder();
+            Helpers.InstallTestTemplate(templateLocation, _log, workingDirectory, home);
+
+            var commandResult = new DotnetNewCommand(_log, templateName, "--allow-scripts", "yes")
+                .WithCustomHive(home)
+                .WithWorkingDirectory(workingDirectory)
+                .Execute();
+
+            commandResult
+                .Should()
+                .ExitWith(0)
+                .And.NotHaveStdErr()
+                .And.HaveStdOutContaining($"The template \"{templateName}\" was created successfully.")
+                .And.NotHaveStdOutContaining("This line goes to stdout")
+                .And.NotHaveStdOutContaining("Run 'chmod +x *.sh'")
+                .And.NotHaveStdOutContaining("Manual instructions: Run 'setup.cmd'")
+                .And.NotHaveStdOutContaining("Manual instructions: Run 'setup.sh'");
+        }
+
+        [Fact]
+        public void RunScript_RedirectOnError()
+        {
+            string templateLocation = "PostActions/RunScript/RedirectOnError";
+            string templateName = "TestAssets.PostActions.RunScript.RedirectOnError";
+            string home = TestUtils.CreateTemporaryFolder("Home");
+            string workingDirectory = TestUtils.CreateTemporaryFolder();
+            Helpers.InstallTestTemplate(templateLocation, _log, workingDirectory, home);
+
+            var commandResult = new DotnetNewCommand(_log, templateName, "--allow-scripts", "yes")
+                .WithCustomHive(home)
+                .WithWorkingDirectory(workingDirectory)
+                .Execute();
+
+            commandResult
+                .Should()
+                .Fail()
+                .And.HaveStdOutContaining($"The template \"{templateName}\" was created successfully.")
+                .And.HaveStdErrContaining("Command failed")
+                .And.HaveStdErrContaining("This line goes to stdout")
```

Comment:
Should this be `NotHaveStdErrContaining`? If that line goes to stdout...

Reply:
No, it's correct. 
The idea is: if the redirect is enabled and script succeeds we won't see the output, but if the script fails it will print both stdout and stderr to as error details (to error stream).